app/src/main/java/unipd/se18/ocrcamera/ImageProcessing.java

```diff
@@ -0,0 +1,349 @@
+        //The third parameter contains additional information that is unused
+
+
+        /*
+            EXPERIMENTAL:
+            finds the text rectangle with the largest area
+            and saves it in "max_contour"
+         */
+        double maxArea = 0;
+        MatOfPoint max_contour = new MatOfPoint();
+        Iterator<MatOfPoint> iterator = contours.iterator();
+        while (iterator.hasNext()) {
+            MatOfPoint contour = iterator.next();
+            double area = Imgproc.contourArea(contour);
+            if (area > maxArea) {
+                maxArea = area;
+                max_contour = contour;
+            }
+        }
+
+        //Creates a rotated rectangle based on "max_contour"
+        RotatedRect rect = Imgproc.minAreaRect(new MatOfPoint2f(max_contour.toArray()));
+        return rect;
+    }
+
+
+    /**
+     * Converts the matrix into a Bitmap
+     * @param matrix the matrix you want to convert
+     * @return the bitmap corresponding to the matrix
+     * @author Thomas Porro (g1)
+     */
+    public Bitmap conversion(Mat matrix) {
+        Bitmap image = Bitmap.createBitmap(matrix.width(), matrix.height(), Bitmap.Config.ARGB_8888);
+        Utils.matToBitmap(matrix, image);
+        return image;
+    }
+
+
+    /**
+     * Crop the matrix with the given rectangle
+     * @param rectangle the part of the image you want to crop
+     * @param mat the matrix you want to crop
+     * @return a matrix that contains only the rectangle
+     * @author Thomas Porro(g1), Oscar Garrido (g1)
+     */
+    public Mat crop(RotatedRect rectangle, Mat mat) {
+
+        RotatedRect rect;
+
+        //Matrices we'll use
+        Mat rotationMat = new Mat();
+        Mat rotatedImg = new Mat();
+        Mat croppedImg = new Mat();
+
+        //Get angle and size from the bounding box
+        double angle = rectangle.angle;
+        Size rect_size = rectangle.size;
+
+        //Thanks to http://felix.abecassis.me/2011/10/opencv-rotation-deskewing/
+        if (rectangle.angle < -45.) {
+            angle += 90.0;
+            double width = rect_size.width;
+            double height = rect_size.height;
+            rect_size.width = height;
+            rect_size.height = width;
+        }
+
+        //Creates the rotation matrix
+        rotationMat = getRotationMatrix2D(rectangle.center, angle, 1.0);
+
+        //Perform the affine transformation
+        Imgproc.warpAffine(mat, rotatedImg, rotationMat, mat.size(), INTER_CUBIC);
+
+        //Crop the resulting image
+        getRectSubPix(rotatedImg, rect_size, rectangle.center, croppedImg);
+        return croppedImg;
+
+    }
+
+
+    /**
+     * Converts a matrix into a Bitmap and saves it in a local directory
+     * @param matrix the matrix to be converted
+     * @param imageName the name of the file being saved
+     * @author Thomas Porro(g1), Oscar Garrido(g1)
+     */
+    private void save(Mat matrix, String imageName) {
```

Comment:
This method is more for testing purposes because, in the end, we do not want to actually save in the gallery all the images created for our processes, but your idea is still good because we could avoid to hardcode the output directory and make the code cleaner!

Reply:
Decision to save in the cache, so, resolved